### DISCUSSION TITLE
Thanks for your hint and a question

### DISCUSSION THREAD

**Discussion by CePeU** [mkdocs-publisher/mkdocs-publisher#77] [Q&A] [answered]
Hi, you commented on the mkdocs issue in april (I just restarted some documentation stuff with mkdocs and saw the notice now on my github). You tool looks really nice and I will take a look allthough I am not yet sure when I will find time.

While reading your setup documentation one question came up for me. Why do you need the Obsidian vault to be in the docs directory? I understand that this probably is the easiest place (after all your python script will know where to look) but one reason for an obsidian plugin with regex for me was that I could use Obsidian totally independently of MkDocs. I can copy my vault anywhere I like, I can setup sync with a remote server etc. Once I have my documentation ready I export it into the MkDocs folder and use MkDocs to generate a static site, preview things and then export them to github.

Would it not be possible for your python setup to have a config file where one could declare the vault location?

**Comment by mkusz** [comment, marked as answer]
Hi @CePeU,
I will have to finally find time to write it down, because it seems to be not clear.
Let's clarify some things:
- your vault can be put inside any directory that is configured inside `mkdocs.yml` as a value of `docs_dir`
- `docs` dir is an output directory (defined by `site_dir` in `mkdocs.yml`) and is setup this way because GitHub uses this specific directory to take files for Github Pages from the same repo where `.md` files are placed
- all publisher plugins are using values from `mkdocs.yml` file, so you can set things up to your needs.
Configuration of the project repository is set up in a very specific way, so any push to git repository is triggering automatically Github Actions that automatically generate documentation output and push it to the web page.
There is a bunch of possibilities how to configure publication. I have done it the way that is fully automatic inside a single repository.
